### ISSUE TITLE
choose download location

### ISSUE THREAD

**Issue by fferraro87** [tympanix/Electorrent#184]
**Client:**
- [ ] µTorrent
- [X ] qBittorrent
- [ ] Transmission
- [ ] rTorrent
- [ ] Deluge

**Client Version:**
v4.2.5 qBittorrent
**Operating System:**
Centos 8
**Application Version:**

**Description:**
Hi, 
when i download a torrent, can i choose download folder location? or move after it download a torrent?

**Comment by tympanix** [comment]
Torrents will download in the default directory. That means, there are no advanced options when adding a torrent. If this is a feature in qBittorrent Web UI, then Electorrent would also be able to implement that feature.

**Comment by fferraro87** [comment]
Yes that's a feature that is already present in qBittorrent Web UI but it isn't supported on electorrent right now

**Comment by fferraro87** [comment]
i see on torrentq.js data.save_path and that's maybe folder location right? @tympanix

**Comment by tympanix** [comment]
Yes and no. That is the save location, yes, but it is information coming from the api and not something you can chose or edit. If you wish to have a look at the implementation have a look at https://github.com/tympanix/node-qbittorrent

**Comment by reconman** [comment]
https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-(qBittorrent-4.1)#add-new-torrent specifies that the parameter `savepath` can be used.

Now it should be possible to specify the location.

**Comment by DynaZor** [comment]
+1

I would really love to have the torrent dialog implemented in Electorrent!
Currently I can't choose what parts of the torrents I want unless I use QB's web interface.

Any updates on the subject matter?

**Comment by tympanix** [comment]
No progress yet. I would want this implementation to be somewhat generic, so that other clients could benefit from this.

I have a few hours a week (if at all) to work on features. At the same time the Electorrent code is getting older by the day and needs tidying up.

**Comment by DynaZor** [comment]
> 
> 
> No progress yet. I would want this implementation to be somewhat generic, so that other clients could benefit from this.
> 
> I have a few hours a week (if at all) to work on features. At the same time the Electorrent code is getting older by the day and needs tidying up.

I would love to help!
What should I learn so I can succeed in making this work?
(I'm a game dev, specifically Tech Artist and programmer in C#)

**Comment by tympanix** [comment]
The technologies used are Javascript, HTML, CSS/Less, AngularJS, NodeJS and the Electron framework. PRs are appreciated. The hardest part about this implementation is that it should be rather generic so that every client could take advantage of the functionality, not just qBittorrent.

**Comment by swannie-eire** [comment]
I would like this feature also for Qbittorrent

**Comment by dansauve** [comment]
I would like the feature also for Transmission. Thanks

**Comment by reconman** [comment]
@dansauve I would recommend using https://github.com/transmission-remote-gui/transgui, since it's made for Transmission and works great.

**Comment by dansauve** [comment]
> @dansauve I would recommend using https://github.com/transmission-remote-gui/transgui, since it's made for Transmission and works great.

Thanks for the reco. I've been using that one for years but I get weird hangs in it all the time when adding torrents... the UI just locks up for 30-120 seconds and usually comes back, but it's annoying. I always thought it was just the client (Transgui) itself but wonder then if it has to do with my server? Transmission runs fine but it's an old Synology DS212+ so it's slow hardware and usually have ~200 torrents going in seeding... Anyway, I thought it was Transgui and that's why I've been looking for an alternative (not to mention Electorrent has a nicer, newer looking GUI ;)

**Comment by tympanix** [comment]
Resolved for v2.10.0 by #507 (choose a save location when adding), #510 (set the location of existing torrents), and #524 (saved reusable locations).
